### DISCUSSION TITLE
Wasapi - How to find the minimum supported `audio.period-size` to prevent crashes?

### DISCUSSION THREAD

**Discussion by jonblk** [FluidSynth/fluidsynth#1344] [Q&A] [answered]
In wasapi `exclusive mode=true` if I set the `audio.period-size` below the minimum the process crashes.

Is this minimum value dynamically generated?  From what I can see it seems to vary depending on the sample rate. 

Is there a way to determine these `audio.period-size` minimum values for wasapi?  

Thanks!

**Comment by derselbst** [comment, marked as answer]
What do you mean by the "process crashes"? The minimum period-size depends on the sampling rate, the audio device and possibly other factors.  If it's not supported it should print an error, possibly even telling you what the minimum period size is.

**Reply by jonblk** [reply]
At the time I had assumed the process had crashed.   

Based on your answer my app now listens for the minimum period-size error, and then opens a new process with the given value. 

Thanks
